Fix build failure with old compilers like gcc 4.2 and clang 3.4.
https://github.com/skvadrik/re2c/pull/292
--- src/dfa/closure_posix.h.orig	2019-07-13 05:08:20.000000000 -0500
+++ src/dfa/closure_posix.h	2020-07-04 08:19:09.000000000 -0500
@@ -125,7 +125,8 @@
     // init: push configurations ordered by POSIX precedence (highest on top)
     state.clear();
     std::sort(reach.begin(), reach.end(), cmp_gor1_t<ctx_t>(ctx));
-    for (typename ctx_t::rcconfiter_t c = reach.rbegin(); c != reach.rend(); ++c) {
+    typename ctx_t::rcconfiter_t c = reach.rbegin(), e = reach.rend();
+    for (; c != e; ++c) {
         nfa_state_t *q = c->state;
         if (q->clos == NOCLOS) {
             q->clos = static_cast<uint32_t>(state.size());
